File: src/lib/scala/Slate.Common/src/main/scala/slate/common/Vars.scala
/**
<slate_header>
  author: Kishore Reddy
  url: https://github.com/kishorereddy/scala-slate
  copyright: 2015 Kishore Reddy
  license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  desc: a scala micro-framework
  usage: Please refer to license on github for more info.
</slate_header>
  */

package slate.common

import scala.collection.mutable.ListBuffer

class Vars(items:Option[List[(String,Any)]]) extends ListMap[String,Any]{

  init(items)


  private def init(items:Option[List[(String,Any)]]):Unit = {
    items.fold(Unit)( all => {
      all.foreach( item => {
        this.add(item._1, item._2)
      })
      Unit
    })
  }
}


object Vars {

  def apply(text:String): Vars = {
    val data = Strings.splitToMapWithPairs(text)
    val buf = ListBuffer[(String,Any)]()
    data.foreach( p => buf.append(( p._1, p._2.asInstanceOf[Any])))
    new Vars(Some(buf.toList))
  }

}

File: src/lib/kotlin/slatekit-common/src/main/kotlin/slatekit/common/Vars.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.common


/**
 * Short hand representation of a list of "variables" backed
 * by the ListMap<String,Any> which represents both a List and Map
 * and provides lookup by both the key and index/iteration like a list
 */
class Vars(items: List<Pair<String, Any>>) : ListMap<String, Any>(items) {

    companion object {

        /**
         * converts a text of "a=1,b=2,c=3" into a Vars object
         * of
         *  - "a" -> 1
         *  - "b" -> 2
         *  - "c" -> 3
         *
         * with key=String, value=Any and separators expected to
         * be "," for pairs values and "=" between key / value
         */
        fun apply(text: String): Vars {
            return if (text.isNullOrEmpty()) {
                Vars(listOf())
            }
            else {
                val data = text.splitToMapWithPairs()
                val buf = mutableListOf<Pair<String, Any>>()
                data.forEach { p -> buf.add(Pair(p.key, p.value as Any)) }
                Vars(buf.toList())
            }
        }
    }
}
